### PR TITLE
[IMP] website: use the user language for the website interface.

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
This commit allows the website to use the user language for parts of
the user interface.

task-2070455

### CODE REVIEW

**Comment by ThanhDodeurOdoo** [comment]
@qsm-odoo not sure about naming of templates, I tried to rename to minimise the diff (if it was often t-called, the t-called template kept the name, if it was often inherited, the inherited/translated template kept the name).

File: addons/web_editor/views/snippets.xml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 <template id="snippets">
+    <t t-call="web_editor.snippets_content" t-lang="env.user.lang"/>
+</template>
+
+<template id="snippets_content">
```
Comment (qsm-odoo) [inline addons/web_editor/views/snippets.xml]: I don't see why you need to do that for the snippet panel rendering in fact. Those templates are rendered via a `render_template` call, can't you just use to user language there ?

Comment (ThanhDodeurOdoo) [reply addons/web_editor/views/snippets.xml]: The problem is that the lang of the context is the wrong one (the default website lang) and the session.user_context doesn't have lang.

I could do a rpc using the user id and getting his lang but that would add a query. What do you think?

File: addons/web_editor/static/src/js/editor/snippets.editor.js [hunk 2]
```diff
@@ -984,6 +985,18 @@ var SnippetsMenu = Widget.extend({
...
+     */
+    async willStart() {
+        await this._super(...arguments);
+        const users = await this._rpc({
+            model: 'res.users',
+            method: 'read',
+            args: [[session.user_id], ['lang']],
+        });
+        this._userLanguage = users[0].lang;
+    },
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/editor/snippets.editor.js]: Why changing this https://github.com/odoo/odoo/blob/13.0/addons/web/models/ir_http.py#L71 instead would not work?

**Comment by ThanhDodeurOdoo** [comment]
@qsm-odoo The parts of the UI that are static templates are still translated (in this case, `website.editorbar`)

For the static templates, I also assumed that the translation was loaded only when requested but it seems that translations for static are loaded all at once there:  https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/core/translation.js#L53-L76

**Comment by qsm-odoo** [comment]
@ThanhDodeurOdoo 

> @qsm-odoo The parts of the UI that are static templates are still translated (in this case, `website.editorbar`)

The left panel was still in french in my example (at least the text that is indeed already translated in that language in Odoo).

> For the static templates, I also assumed that the translation was loaded only when requested but it seems that translations for static are loaded all at once there: https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/core/translation.js#L53-L76

Are you sure? I am debugging in the `add_template` method of `qweb` and I receive the templates directly translated from the server...

**Comment by ThanhDodeurOdoo** [comment]
@qsm-odoo  It seems so, when I debug at https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/core/translation.js#L73

I can alter the translation, for example:
`trans.modules.website.messages[7]` is `{id: 'save', string:'sauvegarder'}` and doing `trans.modules.website.messages[7].string = ".....";` results in having the topbar's "save" button have '.....' as text.

qweb.add_template happens after that, I think that it uses those translations to build them

**Comment by ThanhDodeurOdoo** [comment]
@qsm-odoo although when I try french website, english user it looks like it's working

screenshot-1925569-master-website-ui-language-tso-7a7f04 runbot35 odoo com-2020 04 01-12_03_36

when I try english website, french user, the top right "discard/save" are still in website language because it's in a static template but the left panel has parts that are translated and in this case I think that the non translated parts are just because translations are missing.

screenshot-1925569-master-website-ui-language-tso-7a7f04 runbot35 odoo com-2020 04 01-12_08_25

**Comment by qsm-odoo** [comment]
@ThanhDodeurOdoo Strange, maybe I did not test correctly, my bad.

**Pull request closed without merging**